### PR TITLE
introducing routing with React Router

### PR DESCRIPTION
**Pull request by SergiuB**
- I have "format on save" in the editor, so a few one-liners got split, let me know if I should revert this
- in app.jsx I completely removed the route method, and because of this the diffing sucks

### CODE REVIEW

File: src/index.html [hunk 1]
```diff
@@ -18,5 +19,5 @@
 <script src="//cdnjs.cloudflare.com/ajax/libs/ckeditor/4.4.5/ckeditor.js"></script>
 <script src="//cdnjs.cloudflare.com/ajax/libs/speakingurl/2.0.0/speakingurl.min.js"></script>
 <script src="//cdnjs.cloudflare.com/ajax/libs/moment.js/2.10.2/moment.min.js"></script>
-<script src="https://cdn.rawgit.com/Hacker0x01/react-datepicker/e5a67bd4/dist/react-datepicker.min.js"></script>
-<script src="app.js?{{t}}"></script>
```
Comment (mgax) [inline src/index.html:22]: Why remove `{{t}}`? That was for cache busting.

**Comment by mgax** [comment]
> I have "format on save" in the editor, so a few one-liners got split, let me know if I should revert this

Yup, please revert it.

File: src/app.jsx [hunk 2]
```diff
@@ -153,12 +64,22 @@ class App extends React.Component {
 }
 
 $.get('config.json', (config) => {
+  let Router = ReactRouter.Router,
+    Route = ReactRouter.Route, browserHistory = ReactRouter.browserHistory
```
Comment (mgax) [inline src/app.jsx]: Please use a separate `let` for `Route`.

File: src/index.html (hunk 1 above)
Comment (SergiuB) [inline src/index.html:22]: Good catch, I had a problem at some point, removed it to test smth, then forgot about it

File: src/app.jsx [hunk 3]
```diff
@@ -153,12 +64,22 @@ class App extends React.Component {
...
+
   var query = parseQuery(window.location.search);
   if(config.piwik && ! query['code']) {
     trackPiwik(config.piwik);
   }
-  window.app = React.render(
-    <App config={config} query={query} />,
+  window.__app_config = config;
```
Comment (mgax) [inline src/app.jsx:75]: We should probably refactor the configuration, pass it via `props` all the time. But after we get this pull request merged.

File: src/ckeditor.jsx [hunk 4]
```diff
@@ -30,9 +30,9 @@ class CKEditor extends React.Component {
     var node = React.findDOMNode(this.refs.ck);
     $(node).text(this.toHtml(content));
 
-    var imageBrowserUrl = window.location.href
-      + '&ckImageBrowser=on'
-      + '&siteUrl=' + encodeURIComponent(this.getSiteUrl());
+    var imageBrowserUrl = window.location.pathname + '/ckImageBrowser'
+      + window.location.search + (window.location.search ? '&' : '?')
```
Comment (mgax) [inline src/ckeditor.jsx:34]: If you append `/ckImageBrowser` to the URL, you're already assuming there is no querystring, so you might as well append `?` directly.

Comment (SergiuB) [inline src/ckeditor.jsx:34]: location.pathname does not contain the query string, only the path. QS is in location.search.
It's possible that ?demo will be already present at this time, that's why all this.

File: src/app.jsx (hunk 3 above)
Comment (SergiuB) [inline src/app.jsx:75]: sounds good

**Comment by mgax** [comment]
On master, the markdown editor shows the source code, and the file list has no bullets.

master

On the pull request, there is no markdown editor (jsut the preview), and the file listing has bullets.

pr

**Comment by mgax** [comment]
Hai să merge-uim. Pot să te rog să faci un mare squash, mai puțin ultimul commit?

**Pull request closed without merging**